sql/core.go

```diff
@@ -100,24 +95,37 @@ type Aggregation interface {
 // Node is a node in the execution plan tree.
 type Node interface {
 	Resolvable
-	Transformable
 	fmt.Stringer
 	// Schema of the node.
 	Schema() Schema
 	// Children nodes.
 	Children() []Node
 	// RowIter produces a row iterator from this node.
 	RowIter(*Context) (RowIter, error)
+	// WithChildren returns a copy of the node with children replaced.
```

Comment:
Maybe "transformed" would be better here instead of replaced?

Reply:
Children may or may have not been transformed. This creates a new node with the new children in the places where previous children used to be.

In Opaque the docs refer to "transform" because a node that is opaque will not have its children transformed in TransformUp, TransformExpressions, etc. WithChildren and Children do not have anything to do with that.

On one hand we have the transform concept, which uses Children to get the children, then transforms them and replaces the old with the new ones using WithChildren.
On the other hand, we have replacement, which is what WithChildren is all about.